executor/write.go

```diff
@@ -1167,17 +1167,29 @@ func (e *InsertValues) getRows(cols []*table.Column, ignoreErr bool) (rows [][]t
 			return nil, errors.Trace(err)
 		}
 		e.rowCount = uint64(i)
-		rows[i], err = e.getRow(cols, list, ignoreErr)
+		rows[i], err = e.getRow(cols, list, i, ignoreErr)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
 	}
 	return
 }
 
+func (e *InsertValues) handleErr(col *table.Column, rowIdx int, err error, ignoreErr bool) error {
+	if err == nil {
```

Comment:
No need to check this?

Reply:
Check `err == nil` is ok here. Not a big deal.